sklearn/utils/_typedefs.pxd

```diff
@@ -17,6 +15,7 @@
 ctypedef unsigned char uint8_t
 ctypedef unsigned int uint32_t
 ctypedef unsigned long long uint64_t
+# Note: As of numpy 2.0, np.intp = Py_ssize_t.
```

Comment:
> Instead one should rather use sized integer types such as int32 or uint32 when we know that the number of elements to index is not larger to 2 or 4 billions.

Do you mean `int64` instead of `uint32` for sized integer types?

Reply:
No, because int64 would be too large to index arrays on 32-bit runtimes such as pyodide.

Reply:
The above remark only apply when the integer values stored in the fitted attribute datastructure are used to index an array. This is typically the case for child node ids in decision trees or kd-trees or ball-trees (I believe, I did not check).

Reply:
I‘ll edit the suggestion above in order to commit. (I‘m short of a laptop now - a good thing for a limited while.)